### PR TITLE
[WNMGDS-3126] Adds test coverage for fetching async data with `ds-autocomplete`

### PR DESCRIPTION
**Pull request by kim-cmsds**
## Summary

- [WNMGDS-3126](https://jira.cms.gov/browse/WNMGDS-3126)
- Adds a new story, unit test, and updated snapshots that demonstrates what happens when fetching async data and while supplying a default list of options to `<ds-autocomplete />`

## How to test

1. Pull down the branch, run storybook, and take a look at the "Async Items" [story](http://localhost:6006/?path=/story/web-components-ds-autocomplete--async-items)

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.
- [x] Selected appropriate release milestone

### If this is a change to code:

- [x] Created or updated unit tests to cover any new or modified code
- [x] Verified that running both `yarn test:unit` and `yarn test:browser:all` were each successful
- [x] If necessary, updated unit-test snapshots (`yarn test:unit:update`) and browser-test snapshots (`yarn test:browser:all:update`)

### CODE REVIEW

File: packages/design-system/src/components/web-components/ds-autocomplete/ds-autocomplete.stories.tsx [hunk 1]
```diff
@@ -293,23 +297,107 @@ export const LabeledList = {
 //   },
 // };
 
-export const LoadingMessage = {
+export const LoadingMessage: Story = {
   render: Template,
   args: {
-    clearSearchButton: false,
-    loading: true,
+    'clear-search-button': 'false',
```
Comment (pwolfert) [inline packages/design-system/src/components/web-components/ds-autocomplete/ds-autocomplete.stories.tsx:303]: I see that this was just fixing a mistake in the arg definitions, but is there a special reason we want to disable the clear button here that relates to functionality? Or is it just showing a different variation of the component?

File: packages/design-system/src/components/web-components/ds-autocomplete/__snapshots__/ds-autocomplete.test.tsx.snap [hunk 2]
```diff
@@ -20,22 +20,22 @@ exports[`Autocomplete can add a custom class name to the error message 1`] = `
...
+          id="autocomplete--140__label"
         >
           autocomplete label
         </label>
         <input
           aria-autocomplete="list"
-          aria-controls="autocomplete--133__menu"
-          aria-describedby="autocomplete--133__error"
+          aria-controls="autocomplete--140__menu"
+          aria-describedby="autocomplete--140__error"
```
Comment (pwolfert) [inline packages/design-system/src/components/web-components/ds-autocomplete/__snapshots__/ds-autocomplete.test.tsx.snap:31]: Could this test benefit from a statically defined id as well with `root-id`?

File: packages/design-system/src/components/web-components/ds-autocomplete/ds-autocomplete.test.tsx [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import { render, waitFor, screen } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import { config } from '../../config';
 import userEvent from '@testing-library/user-event';
 import './ds-autocomplete';
 
 const defaultItems = JSON.stringify([{ id: 'kRf6c2fY', name: 'Cook County, IL' }]);
+const updatedItems = JSON.stringify([{ id: 'Yf2c6fRk', name: 'Marion County, OR' }]);
```
Comment (pwolfert) [inline packages/design-system/src/components/web-components/ds-autocomplete/ds-autocomplete.test.tsx:7]: 👍

**Review by pwolfert — APPROVED** [review]
Test looks good 👍

File: packages/design-system/src/components/web-components/ds-autocomplete/ds-autocomplete.stories.tsx (hunk 1 above)
Comment (kim-cmsds) [reply packages/design-system/src/components/web-components/ds-autocomplete/ds-autocomplete.stories.tsx:303]: There are already several stories that have the clear search button included by default, and this change "fixes" the intended behavior of this story, which is to display the ds-autocomplete component without the clear search button. In other words, there's no special reason to disable the clear button.

File: tests/browser/snapshots/stories/web-components-ds-autocomplete--no-results-healthcare--Mobile-Chrome.png
Comment (kim-cmsds) [inline tests/browser/snapshots/stories/web-components-ds-autocomplete--no-results-healthcare--Mobile-Chrome.png:1]: @pwolfert here's the correct version of the snapshot now that the arg (`clear-search-button`) was passed in correctly

Comment (pwolfert) [reply tests/browser/snapshots/stories/web-components-ds-autocomplete--no-results-healthcare--Mobile-Chrome.png:1]: 💯

**Review by tamara-corbalt — APPROVED** [review]
LGTM

**Review by jack-ryan-nava-pbc — APPROVED** [review]
Beautful!

**Pull request merged**